src/debugging/coreclr/CliDockerClient.ts

```diff
@@ -264,18 +264,18 @@ export class CliDockerClient implements DockerClient {
         return id.substring(0, 12);
     }
 
-    public async exec(containerNameOrId: string, args: string, options?: DockerExecOptions): Promise<string> {
-        options = options || {};
+    public async exec(containerNameOrId: string, args: string, dockerExecOptions?: DockerExecOptions, processExecOptions?: ProcessProviderExecOptions): Promise<string> {
```

Comment:
I'm not a fan of pushing the `ProcessProviderExecOptions` through the Docker client (as that's a leaky abstraction--who says the Docker client implementation even uses child processes as opposed to, say, Dockerode?).

What's the underlying issue?  Is it that `processProvider.exec()` is throwing on failure rather than returning the `stdout`?  If we don't want to change the semantics of `dockerClient.exec()` (in that it throws on error rather than return the `stdout`) then I think we should add an optional `progress` property to the `DockerExecOptions` for callers which want the `stdout` up to the point of failure.

Reply:
Yes the issue is that processProvider.exec() throws on failure. I will make the progress as property of DockerExecOptions